packages/design-system/src/components/Accordion/AccordionItem.tsx

```diff
@@ -49,98 +49,86 @@ export interface AccordionItemProps {
    */
   openIcon?: React.ReactNode;
 }
+
 export interface AccordionItemState {
   isOpen?: boolean;
 }
-export class AccordionItem extends React.Component<AccordionItemProps, AccordionItemState> {
-  static defaultProps = {
-    headingLevel: '2',
-    closeIcon: (
-      <RemoveIcon
-        className="ds-c-accordion__button-icon"
-        title={t('accordion.close')}
-        ariaHidden={false}
-      />
-    ),
-    openIcon: (
-      <AddIcon
-        className="ds-c-accordion__button-icon"
-        title={t('accordion.open')}
-        ariaHidden={false}
-      />
-    ),
-  };
-
-  buttonId: string;
-  contentId: string;
-  isControlled: boolean;
 
-  constructor(props: AccordionItemProps) {
-    super(props);
-
-    this.isControlled = !!props.onChange;
-    this.state = this.isControlled ? {} : { isOpen: !!props.defaultOpen };
-    this.handleClick = this.handleClick.bind(this);
-    this.contentId = props.id || uniqueId('accordionItem_');
-    this.buttonId = `${this.contentId}-button`;
-  }
+export const AccordionItem: React.FC<AccordionItemProps> = ({
+  buttonClassName,
+  children,
+  contentClassName,
+  defaultOpen = false,
+  heading,
+  headingLevel = '2',
+  id,
+  isControlledOpen,
```

Comment:
For a future update, maybe we should make a note to change the name of this to just `isOpen`. It doesn't conflict with anything that I can tell. We could add that prop and deprecate the old one for a while.